sdk/java-sdk-protobuf-testkit/src/main/scala/kalix/javasdk/testkit/impl/EventingTestKitImpl.scala

```diff
@@ -397,6 +398,7 @@ private[testkit] object TestKitMessageImpl {
     val bt = BoxedType(clazz)
     payload match {
       case m if bt.isInstance(m) => m.asInstanceOf[T]
+      case m: String             => JsonSupport.getObjectMapper.readerFor(clazz).readValue(m)
```

Comment:
this is the fix.. it can only be unwrapped at this point when the class type is known